### PR TITLE
[Issue394]Add better documents for Sum, Mean, JoinTable and Padding

### PR DESCRIPTION
**Pull request by psyyz10**

### CODE REVIEW

File: dl/src/main/scala/com/intel/analytics/bigdl/nn/JoinTable.scala [hunk 1]
```diff
@@ -27,6 +27,11 @@ import scala.reflect.ClassTag
  * It is a table module which takes a table of Tensors as input and
  * outputs a Tensor by joining them together along the dimension `dimension`.
  *
+ * Note: the input tensor can have indefinite number of dimensions,
+ * so modules can not automatically inference whether the user use the batch or not
+ * and what the actual dimensions of input excluding the batch dimension.
+ * Hence `nInputDims` represents the actual dimensions of given input excluding the batch.
+ *
```
Comment (jason-dai) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/JoinTable.scala]: This is somewhat confusing. Just say somehting like "the input to this layer is expected to a tensor, or a batch of tensors; when using mini-batch, a batch of tensors will be passed to the layer and the user need to specify the number of dimensions of each input tensor in the batch using nInputDims"

File: dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala [hunk 2]
```diff
@@ -228,6 +228,19 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
...
+   * otherwise operation will base on no-batch assumption (with pure online stochastic
+   * gradient descent)
+   */
+  protected var batch: Boolean = false
+
+  def isBatchMode: Boolean = batch
+
+  def setBatchMode(useBatch: Boolean): Unit =
+    batch = useBatch
+
```
Comment (jason-dai) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala]: I don't think we need this for now.

Comment (psyyz10) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala]: Removed.

File: dl/src/main/scala/com/intel/analytics/bigdl/nn/JoinTable.scala (hunk 1 above)
Comment (psyyz10) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/JoinTable.scala]: Yes, I agree. Have changed, please check

Comment (jason-dai) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/JoinTable.scala]: You may want to change the scala doc for the `nInputDims` parameter accordingly too.

**Comment by yiheng** [comment]
Please merge these commits into one before you push to the repo

**Pull request merged**